docs/adr/002-application-architecture-for-configurability.md

```diff
@@ -0,0 +1,117 @@
+# Application Architecture for Configurability
+
+* Status: **In Review**, In Progress, Done
+* Authors: @j08lue, @sandrahoang686, @faustoperez
+* Deciders: @j08lue, @faustoperez, @hanbyul-here
+* As of: 2024/03
+
+## Context and Problem Statement
+
+VEDA UI 1.0 was built as a reusable white-label application for independent science data and information projects. These projects should be able to stand up and configure their own instance with their content. The assumption was that projects would reuse the entire application and change mostly the styling and scalable content items such as datasets and stories, and would only need to change a few other elements. The application (pages, functional components; VEDA UI) is separate from content (VEDA config) and pages and elements can be changed via component overrides.
+
+In 2023, our team supported two new projects to integrate VEDA UI for their needs: the U.S. Greenhouse Gas Center (earth.gov/ghgcenter) and the Earth Information Center (EIC; earth.gov). These projects had much wider needs for customization or adaptation, mostly related to the context of the project:
+1. Different thematic categorization of content with hierarchies or tags
+2. Project-specific headers and disclaimers on several pages, also static / functional ones like the data catalog
+3. Changes to navigation - main, secondary, page
+4. Additional pages
+5. New CMS-typical content types like events or announcements
+
+We are expecting new instances to start using VEDA in 2024.
+
+### Challenge 1: Cumbersome customization to the degree required
+Implementing these additional customizations proved to be cumbersome with the current application structure, because they require page overrides and new customization options across the application, which grew fast in complexity.
+
+### Challenge 2: Dependency injection
```

Comment:
Nitpick: May I suggest considering a different title for this challenge? While 'Dependency Injection' is a well-established design pattern, it might be beneficial to explore a title that better reflects the unique aspects of this specific challenge.

Reply:
Good point. Is the updated title better?

> Challenge 2: Unconventional combination mechanism for instance configuration and UI components (via Git)

Reply:
Yeah I think so. Thanks!